extensions/vertx-http/dev-ui-spi/src/main/java/io/quarkus/devui/spi/page/Page.java

```diff
@@ -61,6 +64,12 @@ protected Page(String icon,
 
     public String getId() {
         String id = this.title.toLowerCase().replaceAll(SPACE, DASH);
+        try {
+            id = URLEncoder.encode(id, StandardCharsets.UTF_8.toString());
```

Comment:
Is it used for all pages? I would expect the URL to be stable, regardless of the changes in the title?

Should we provide a clean and stable id instead?

If we don't, maybe we could still use something a bit cleaner and an approach similar to https://github.com/slugify/slugify (without necessarily using this library).

Reply:
Yes. Exactly my thought.

I'm not saying we cannot merge this. But I'm not sure it's the ideal solution. URLs should be stable as much as possible.

Reply:
Let's open a separate issue for that, it's a bigger piece of work